pyproject.toml

```diff
@@ -26,7 +26,7 @@ dependencies = [
 
 [build-system]
 requires = [
-    "nanobind>=1.8.0",
+    "nanobind>=1.8.0,<=2.2.0",
```

Comment:
Do we need this if we upgraded to 2.2 above?

Reply:
Just to be safe. I don't think they provide a stable API for returning ndarray without copy. Maybe we should just ask the upstream.

Reply:
Will it matter anymore if we switch to returning a new array?

Reply:
Still matters, because we still want to return arrays without copying. Returning a new array just avoids reading back the array after passing it to the user, in case numpy did some weird things to the array.

Reply:
Okay, yeah let's ask upstream. I think I'd rather eat the copy cost than rely on a non-stable API.

Reply:
Looking at the code, it seems that we are doing a copy anyway... Maybe we can find a simpler way.